### DISCUSSION TITLE
Need Help With Jekyll GitHub Project Page and Team Git Workflow

### DISCUSSION THREAD

**Discussion by vinniefalco** [community/community#23657] [Other Feature Feedback, Questions, & Ideas]
<p>I have published <a href="http://cppalliance.com" rel="nofollow noopener">http://cppalliance.com</a> from <a href="https://github.com/CPPAlliance/cppalliance.github.io" rel="nofollow noopener">https://github.com/CPPAlliance/cppalliance.github.io</a></p>
<p>As you can see, there’s a CNAME. Everything is great, the page renders as it should and when I push changes the pages get rebuilt.</p>
<p>However, multiple people are working on this now. Similarly to how a team might collaborate on building a C++ program, I have made a fork of the repository in my personal account:<br>
<a href="https://github.com/vinniefalco/cppalliance" rel="nofollow noopener">https://github.com/vinniefalco/cppalliance</a></p>
<p>Now I can make changes, rebase, squash, fold, and generally arrange the branch any way that I want before submitting it as a pull request to the upstream repository. A problem arises, how to preview the changes before the merge? Well, I can visit the Project GitHub Page:<br>
<a href="https://vinniefalco.github.io/cppalliance/" rel="nofollow noopener">https://vinniefalco.github.io/cppalliance/</a></p>
<p>I can see some of my content, but where did the style sheet go? Is this because the URI is not a root target?</p>
<p>I am also getting an error in my email about the CNAME being taken. Well, that makes sense. We can’t have every fork also be aliased to <a href="http://cppalliance.com" rel="nofollow noopener">cppalliance.com</a>. But I also don’t want to get this error every time. I could delete the CNAME file from my fork, but that would create headaches when merging the submitted pull request to the upstream repository. Is there a way to disable the CNAME from my fork’s Project’s settings? I don’t see it - consider adding it.</p>
<p>Anyway, I would very much like to get to a point where the team members can preview change to their individual forks before submitting the pull request, using the wonderful GitHub Project Pages mechanism described in the docs.</p>
<p>Any advice?</p>
<p>Thanks</p>

**Comment by darrowby385** [comment, marked as answer]
<p>Hi Vinnie!</p>
<p>I’m afraid to say that GitHub Pages sites with custom domains don’t play especially well with forks at the moment. For now, there isn’t a way to supress the CNAME warnings, other than configuring an email filter on your end to automatically archive these messages. It’s not the most elegant solution, but it does work.</p>
<p>Another alternative is to bypass publishing the fork altogether, and instead use a local Jekyll environment to test changes offline. We have documentation here on how to get that set up:</p>
<p><a href="https://help.github.com/articles/setting-up-your-github-pages-site-locally-with-jekyll/" rel="nofollow noopener">https://help.github.com/articles/setting-up-your-github-pages-site-locally-with-jekyll/</a></p>
<p>Of course, we’re aware that this could be better, and it is something that we’d like to improve in the future. I can’t make any promises, but I’ll certainly communicate your observations to our team!</p>
<p>Finally, there’s the CSS issue you mentioned – this isn’t specific to GitHub Pages, but is rather a general HTML concept. At the moment, the CSS file is referenced as:</p>
<pre><code>href="/css/main.css"
</code></pre>
<p>The leading slash before “css” there represents the root of the published URL. This works just fine when the site is published at <a href="http://cppalliance.com" rel="nofollow noopener">http://cppalliance.com</a>, but causes trouble when published at <a href="https://username.github.io/cppalliance.github.io" rel="nofollow noopener">https://username.github.io/cppalliance.github.io</a>. That’s because the slash will end up referring to <a href="https://username.github.io" rel="nofollow noopener">https://username.github.io</a> (the root of the domain), not the relevant “<a href="http://cppalliance.github.io" rel="nofollow noopener">cppalliance.github.io</a>” subdirectory, where the CSS file is actually stored.</p>
<p>The simplest solution to this trouble is to remove the leading slash, leaving you with the following:</p>
<pre><code>href="css/main.css"
</code></pre>
<p>This construction tells the browser to look for a “css” folder within whatever directory the index.html page is being published from, and should therefore function perfectly regardless of which domain the site is published at. You can learn more about HTML file paths here:</p>
<p><a href="https://www.w3schools.com/html/html_filepaths.asp" rel="nofollow noopener">https://www.w3schools.com/html/html_filepaths.asp</a></p>
<p>I hope that helps!</p>

**Comment by PsychoData** [comment]
<p>The best thing I have found so far is to</p>
<ul>
<li>Fork the repo,</li>
<li>Change the CNAME right away
<ul>
<li>to a domain you have access to like “{NewCname}.PersonalDomain.com”</li>
<li>Configure your Domain’s public DNS and ensure Github Pages is enabled in the Settings of your Forked Repo
<ul>
<li>or Delete the CNAME file entirely by itself</li>
</ul>
</li>
<li>after it finishes deploying, access at your new Cname through https:// NewCname.PersonalDomain .com or https:// github .com/ForkedUserName/RepoName depending on if you changed or deleted the CNAME</li>
</ul>
</li>
<li>cherry pick the Commits you are including in your pull requests to upstream later on and skip the Commit where you changed the CNAME file, sadly it doesn’t look like there is a way to Cherry Pick in the current Github Desktop or WebUI that I can find</li>
</ul>
<p>Another alternative I was finding was essentially to clear the file from the Index in the fork and skip the file from (https:// stackoverflow .com/a/45021381/2238544) checking for things to commit or other methods to revert automatically ( https:// stackoverflow .com/a/25315991/2238544) but they seem similarly awful</p>
<p>Possibly a pre-receive hook (https:// docs.github .com/en/enterprise-server@2.21/github/collaborating-with-issues-and-pull-requests/working-with-pre-receive-hooks ) looks promising, though I don’t think that would be available on <a href="http://Github.com" rel="noopener nofollow ugc">Github.com</a>, only the enterprise server</p>
<p>Or possibly Compliance Items and/or Webhooks (<a href="https://resources.github" rel="noopener nofollow ugc">https://resources.github</a> .com/whitepapers/practical-guide-to-CI-with-Jenkins-and-GitHub/ ), but I haven’t gotten into that yet, and that might fit best on the Upstream Repo, not just your fork</p>
<p>PS Sorry about the awful links, it was blocking me from posting more than two links since I’m a new user</p>
